scripts/ida_gef.py

```diff
@@ -64,6 +65,95 @@ def ishex(s):
     return s.startswith("0x") or s.startswith("0X")
 
 
+"""
+Wrappers taken from FIRST IDA Plugin
+https://github.com/vrtadmin/FIRST-plugin-ida/blob/dev/first_plugin_ida/first.py
+On IDA > 7.2  some functions are thread-safe and need to be executed on the
+main Thread.
+The function idaapi.execute_sync is called to execute these functions.
+"""
+class IDAWrapper(object):
+    '''
+    Class to wrap functions that are not thread safe.  These functions must
+    be run on the main thread to avoid random crashes (and starting in 7.2,
+    this is enforced by IDA, with an exception being generated if a
+    thread-unsafe function is called from outside of the main thread.)
+    '''
+    mapping = {
+        'get_tform_type' : 'get_widget_type',
+    }
+    def __init__(self):
+        self.version = idaapi.IDA_SDK_VERSION
+
+    def __getattribute__(self, name):
+        default = '[1st] default'
+
+        if (idaapi.IDA_SDK_VERSION >= 700) and (name in IDAWrapper.mapping):
+            name = IDAWrapper.mapping[name]
+
+        val = getattr(idaapi, name, default)
+        if val == default:
+            val = getattr(idautils, name, default)
+
+        if val == default:
+            val = getattr(idc, name, default)
+
+        if val == default:
+            msg = 'Unable to find {}'.format(name)
+            idaapi.execute_ui_requests((FIRSTUI.Requests.Print(msg),))
```

Comment:
We need our own way to error here.

https://www.hex-rays.com/products/ida/support/idapython_docs/ida_kernwin-module.html#execute_ui_requests